csrc/host_ir/host_ir.h

```diff
@@ -117,13 +117,23 @@ class PostOnStream : public Expr {
 
 class Stream : public Val {
  public:
-  Stream(IrBuilderPasskey passkey);
+  // if index is provided, the IR represents the streams whose index is the
+  // dynamic value of that index. Otherwise, it statically represents a new
+  // Stream.
+  Stream(IrBuilderPasskey passkey, Val* index = nullptr);
   Stream(const Stream* src, IrCloner* ir_cloner);
   bool sameAs(const Statement* other) const override;
```

Comment:
Should `sameAs` now return true if indices are the same and are not null?

Reply:
fixed